src/plone/base/interfaces/recyclebin.py

```diff
@@ -0,0 +1,127 @@
+        default=100,
+        min=10,
+    )
+
+    restore_to_initial_state = schema.Bool(
+        title="Restore to initial workflow state",
+        description="When enabled, restored content will be set to its initial workflow state (usually 'draft') instead of the workflow state it was in when deleted.",
+        default=False,
+        required=False,
+    )
+
+
+class IRecycleBin(Interface):
+    """Interface for the recycle bin functionality"""
+
+    def add_item(obj, original_container, original_path):
+        """Add deleted item to recycle bin
+
+        Args:
+            obj: The object being deleted
+            original_container: The parent container before deletion
+            original_path: The full path to the object before deletion
+
+        Returns:
+            The ID of the item in the recycle bin
+        """
+
+    def get_items():
+        """Return all items in recycle bin
+
+        Returns:
+            A list of dictionaries with information about deleted items
+        """
+
+    def get_item(item_id):
+        """Get a specific deleted item by ID
+
+        Args:
+            item_id: The ID of the deleted item in the recycle bin
+
+        Returns:
+            Dictionary with item information or None if not found
+        """
+
+    def restore_item(item_id, target_container=None):
+        """Restore item to original location or specified container
+
+        Args:
+            item_id: The ID of the item in the recycle bin
+            target_container: Optional target container to restore to
+                              (defaults to original container)
+
+        Returns:
+            The restored object or None if restore failed
+        """
+
+    def purge_item(item_id):
+        """Permanently delete an item
+
+        Args:
+            item_id: The ID of the item in the recycle bin
+
+        Returns:
+            Boolean indicating success
+        """
+
+    def purge_expired_items():
+        """Purge items that exceed the retention period
+
+        Returns:
+            Number of items purged
+        """
+
+    def is_enabled():
+        """Check if recycle bin is enabled in the registry settings
+
+        Returns:
+            Boolean indicating whether recycle bin is enabled
+        """
+
+
+class IRecycleBinForm(Interface):
+    """Schema for the recycle bin form"""
+
+    selected_items = schema.List(
+        title="Selected Items",
+        description="Selected items for operations",
+        value_type=schema.TextLine(),
+        required=False,
+    )
+
+
+class IRecycleBinItemForm(Interface):
+    """Schema for the recycle bin item form"""
+
+    target_container = schema.Choice(
+        title="Target container",
+        description="Choose the container where the item should be restored when original location no longer exists",
+        vocabulary="plone.recyclebin.RestoreContainers",
+        required=False,
+    )
```

Comment:
@rohnsha0 Please move IRecyleBinForm and IRecycleBinItemForm into the same module (in CMFPlone) where the form that uses them is.

It's usually nice to keep form schemas next to the form code, since they are closely related.

We make an exception for interfaces that might need to be imported from many packages. That's true for IRecycleBinControlPanelSettings and IRecycleBin (since there are various packages that need to look up settings or the recycle bin utility in order to check if it's enabled), so they should stay here.